File: S/2418-sort-the-people/main.go
package main

import "sort"

func sortPeople(names []string, heights []int) []string {
	m := map[int]string{}

	for i := 0; i < len(names); i++ {
		m[heights[i]] = names[i]
	}
	sort.Sort(sort.Reverse(sort.IntSlice(heights)))
	ret := []string{}
	for _, h := range heights {
		ret = append(ret, m[h])
	}
	return ret
}

File: S/2418-sort-the-people/main.rs
struct Solution {}

impl Solution {
  pub fn sort_people(names: Vec<String>, heights: Vec<i32>) -> Vec<String> {
    let mut a = names
      .into_iter()
      .zip(heights.into_iter())
      .collect::<Vec<(String, i32)>>();
    a.sort_by(|x, y| x.1.cmp(&y.1).reverse());
    a.into_iter().map(|v| v.0).collect::<Vec<String>>()
  }
}
